src/CodeGen_Hexagon.cpp

```diff
@@ -87,7 +87,7 @@ Stmt acquire_hvx_context(Stmt stmt, const Target &target) {
     // register a destructor to call halide_qurt_hvx_unlock.
     Stmt check_hvx_lock = call_halide_qurt_hvx_lock(target);
     Expr dummy_obj = reinterpret(Handle(), cast<uint64_t>(1));
-    Expr hvx_unlock = Call::make(Int(32), Call::register_destructor,
+    Expr hvx_unlock = Call::make(Handle(), Call::register_destructor,
```

Comment:
Here and elsewhere: I don't see that the sig of Call::register_destructor changed; was this just always wrong before and we were just getting lucky? (If so, is change actually related to bfloat or is it just a driveby fix?)

Reply:
This was always wrong before, but there was no checking that the llvm return type of the intrinsic. matched the Halide type. I added checking in this PR to make sure I got the bfloat stuff right, and it necessitated fixes elsewhere.